diff -up OpenBLAS-0.3.10/kernel/power/zdot.c.ppc64le OpenBLAS-0.3.10/kernel/power/zdot.c
--- OpenBLAS-0.3.10/kernel/power/zdot.c.ppc64le	2020-06-14 22:03:04.000000000 +0200
+++ OpenBLAS-0.3.10/kernel/power/zdot.c	2020-09-18 11:18:16.102180677 +0200
@@ -94,8 +94,11 @@ FLOAT _Complex CNAME(BLASLONG n, FLOAT *
 
 	if ( n <= 0 ) 
 	{
+	  /*
 	        __real__ result = 0.0 ;
         	__imag__ result = 0.0 ;
+	  */
+	        result = OPENBLAS_MAKE_COMPLEX_FLOAT(0.0,0.0);
 		return(result);
 
 	}
@@ -149,12 +152,17 @@ FLOAT _Complex CNAME(BLASLONG n, FLOAT *
 	}
 
 #if !defined(CONJ)
+	/*
 	__real__ result = dot[0] - dot[1];
 	__imag__ result = dot[2] + dot[3];
+	*/
+	result = OPENBLAS_MAKE_COMPLEX_FLOAT(dot[0]-dot[1],dot[2]+dot[3]);
 #else
+	/*
 	__real__ result = dot[0] + dot[1];
 	__imag__ result = dot[2] - dot[3];
-
+	*/
+	result = OPENBLAS_MAKE_COMPLEX_FLOAT(dot[0]+dot[1],dot[2]-dot[3]);
 #endif
 
 	return(result);
