diff -up OpenBLAS-0.3.11/cpp_thread_test/Makefile.tests OpenBLAS-0.3.11/cpp_thread_test/Makefile
--- OpenBLAS-0.3.11/cpp_thread_test/Makefile.tests	2020-10-17 22:14:12.000000000 +0200
+++ OpenBLAS-0.3.11/cpp_thread_test/Makefile	2020-10-18 18:18:59.460796184 +0200
@@ -1,13 +1,14 @@
-include ../Makefile.rule
+TOPDIR = ..
+include $(TOPDIR)/Makefile.system
 
 all :: dgemv_tester dgemm_tester
 
 dgemv_tester :
-	$(CXX) $(COMMON_OPT) -Wall -Wextra -Wshadow -fopenmp -std=c++11 dgemv_thread_safety.cpp ../libopenblas.a -lpthread -o dgemv_tester
+	$(CXX) $(COMMON_OPT) -Wall -Wextra -Wshadow -fopenmp -std=c++11 dgemv_thread_safety.cpp ../$(LIBNAME) -lpthread -o dgemv_tester
 	./dgemv_tester
 
 dgemm_tester : dgemv_tester
-	$(CXX) $(COMMON_OPT) -Wall -Wextra -Wshadow -fopenmp -std=c++11 dgemm_thread_safety.cpp ../libopenblas.a -lpthread -o dgemm_tester
+	$(CXX) $(COMMON_OPT) -Wall -Wextra -Wshadow -fopenmp -std=c++11 dgemm_thread_safety.cpp ../$(LIBNAME) -lpthread -o dgemm_tester
 	./dgemm_tester
 
 clean ::
diff -up OpenBLAS-0.3.11/Makefile.tests OpenBLAS-0.3.11/Makefile
--- OpenBLAS-0.3.11/Makefile.tests	2020-10-17 22:14:12.000000000 +0200
+++ OpenBLAS-0.3.11/Makefile	2020-10-18 18:19:54.134437445 +0200
@@ -138,13 +138,13 @@ tests :
 ifeq ($(NOFORTRAN), $(filter 0,$(NOFORTRAN)))
 	touch $(LIBNAME)
 ifndef NO_FBLAS
-	$(MAKE) -C test all
+	$(MAKE) -C test FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 endif
-	$(MAKE) -C utest all
+	$(MAKE) -C utest FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 ifneq ($(NO_CBLAS), 1)
-	$(MAKE) -C ctest all
+	$(MAKE) -C ctest FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 ifeq ($(CPP_THREAD_SAFETY_TEST), 1)
-	$(MAKE) -C cpp_thread_test all
+	$(MAKE) -C cpp_thread_test FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 endif
 endif
 endif
