diff -up serial/Makefile.system_lapack serial/Makefile
--- serial/Makefile.system_lapack	2013-07-20 18:05:36.000000000 +0300
+++ serial/Makefile	2013-07-23 12:28:02.961948050 +0300
@@ -202,74 +202,8 @@ hpl_p :
 	fi; \
 	done
 
-ifeq ($(NO_LAPACK), 1)
-netlib : 
-
-else
-netlib : lapack_prebuild
-ifndef NOFORTRAN
-	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapacklib
-endif
-ifndef NO_LAPACKE
-	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapackelib
-endif
-endif
-
-prof_lapack : lapack_prebuild
-	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapack_prof
-
-lapack_prebuild :
-ifndef NOFORTRAN
-	-@echo "FORTRAN     = $(FC)" > $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "OPTS        = $(FFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "POPTS       = $(FPFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "NOOPT       = $(FFLAGS) -O0" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "PNOOPT      = $(FPFLAGS) -O0" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "LOADOPTS    = $(FFLAGS) $(EXTRALIB)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "CC          = $(CC)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "override CFLAGS      = $(LAPACK_CFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "ARCH        = $(AR)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "ARCHFLAGS   = -ru" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "RANLIB      = $(RANLIB)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "LAPACKLIB   = ../$(LIBNAME)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "LAPACKELIB  = ../$(LIBNAME)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "LAPACKLIB_P = ../$(LIBNAME_P)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "SUFFIX      = $(SUFFIX)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "PSUFFIX     = $(PSUFFIX)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@echo "CEXTRALIB   = $(EXTRALIB)" >> $(NETLIB_LAPACK_DIR)/make.inc
-	-@cat  make.inc >> $(NETLIB_LAPACK_DIR)/make.inc
-endif
-
-lapack-3.4.2 : lapack-3.4.2.tgz
-ifndef NOFORTRAN
-ifndef NO_LAPACK
-	@if test `$(MD5SUM) $< | $(AWK) '{print $$1}'` = 61bf1a8a4469d4bdb7604f5897179478; then \
-		echo $(TAR) zxf $< ;\
-		$(TAR) zxf $< && (cd $(NETLIB_LAPACK_DIR); $(PATCH) -p1 < ../patch.for_lapack-3.4.2) ;\
-		rm -f $(NETLIB_LAPACK_DIR)/lapacke/make.inc ;\
-	else \
-		rm -rf $(NETLIB_LAPACK_DIR) ;\
-		echo "	Cannot download lapack-3.4.2.tgz or the MD5 check sum is wrong (Please use orignal)."; \
-		exit 1; \
-	fi
-endif
-endif
-
-LAPACK_URL=http://www.netlib.org/lapack/lapack-3.4.2.tgz
-
-lapack-3.4.2.tgz :
-ifndef NOFORTRAN
-#http://stackoverflow.com/questions/7656425/makefile-ifeq-logical-or
-ifeq ($(OSNAME), $(filter $(OSNAME),Darwin NetBSD))
-	curl -O $(LAPACK_URL);
-else
-ifeq ($(OSNAME), FreeBSD)
-	fetch $(LAPACK_URL);
-else
-	wget -O $@ $(LAPACK_URL);
-endif
-endif
-endif
+netlib :
+	@$(MAKE) -C $(NETLIB_LAPACK_DIR)
 
 large.tgz : 
 ifndef NOFORTRAN
diff -up serial/Makefile.system.system_lapack serial/Makefile.system
--- serial/Makefile.system.system_lapack	2013-07-20 18:05:36.000000000 +0300
+++ serial/Makefile.system	2013-07-23 12:29:00.084410995 +0300
@@ -9,7 +9,7 @@ ifndef TOPDIR
 TOPDIR = .
 endif
 
-NETLIB_LAPACK_DIR = $(TOPDIR)/lapack-netlib
+NETLIB_LAPACK_DIR = $(TOPDIR)/netliblapack
 
 # Default C compiler
 # - Only set if not specified on the command line or inherited from the environment.
