diff -up OpenBLAS-0.2.19/Makefile.tests OpenBLAS-0.2.19/Makefile
--- OpenBLAS-0.2.19/Makefile.tests	2016-09-01 05:58:42.000000000 +0200
+++ OpenBLAS-0.2.19/Makefile	2016-09-01 22:52:27.092706000 +0200
@@ -110,11 +110,11 @@ tests :
 ifndef NOFORTRAN
 	touch $(LIBNAME)
 ifndef NO_FBLAS
-	$(MAKE) -C test all
-	$(MAKE) -C utest all
+	$(MAKE) -C test FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
+	$(MAKE) -C utest FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 endif
 ifndef NO_CBLAS
-	$(MAKE) -C ctest all
+	$(MAKE) -C ctest FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 endif
 endif
 
