diff -up OpenBLAS-0.3.3/Makefile.rule.tls OpenBLAS-0.3.3/Makefile.rule
--- OpenBLAS-0.3.3/Makefile.rule.tls	2018-08-31 00:07:48.000000000 +0200
+++ OpenBLAS-0.3.3/Makefile.rule	2018-09-29 19:00:47.804678736 +0200
@@ -113,7 +113,7 @@ USE_SIMPLE_THREADED_LEVEL3 = 1
 # thread-local storage instead of a central memory buffer in memory.c
 # Note that if your system uses GLIBC, it needs to have at least glibc 2.21
 # for this to work.
-USE_TLS = 1
+#USE_TLS = 1
 
 # If you want to drive whole 64bit region by BLAS. Not all Fortran
 # compiler supports this. It's safe to keep comment it out if you
