diff -up OpenBLAS-0.2.15/driver/others/memory.c.orig OpenBLAS-0.2.15/driver/others/memory.c
--- OpenBLAS-0.2.15/driver/others/memory.c.orig	2015-10-27 21:44:50.000000000 +0100
+++ OpenBLAS-0.2.15/driver/others/memory.c	2016-01-12 20:22:14.935338309 +0100
@@ -1338,7 +1338,7 @@ static void gotoblas_memory_init(void) {
 
 static int gotoblas_initialized = 0;
 
-void CONSTRUCTOR gotoblas_init(void) {
+void gotoblas_init(void) {
 
   if (gotoblas_initialized) return;
 
@@ -1381,7 +1381,7 @@ void CONSTRUCTOR gotoblas_init(void) {
 
 }
 
-void DESTRUCTOR gotoblas_quit(void) {
+void gotoblas_quit(void) {
 
   if (gotoblas_initialized == 0) return;
 
diff -up OpenBLAS-0.2.15/getarch.c.orig OpenBLAS-0.2.15/getarch.c
