diff -up OpenBLAS-0.2.18/Makefile.orig OpenBLAS-0.2.18/Makefile
--- OpenBLAS-0.2.18/Makefile.orig	2016-04-12 21:29:19.000000000 +0200
+++ OpenBLAS-0.2.18/Makefile	2016-07-13 21:32:17.786003290 +0200
@@ -112,11 +112,11 @@ ifndef TARGET
 ifndef CROSS
 	touch $(LIBNAME)
 ifndef NO_FBLAS
-	$(MAKE) -C test all
-	$(MAKE) -C utest all
+	$(MAKE) -C test FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
+	$(MAKE) -C utest FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 endif
 ifndef NO_CBLAS
-	$(MAKE) -C ctest all
+	$(MAKE) -C ctest FC="$(FC)" CC="$(CC)" COMMON_OPT="$(COMMON_OPT)" FCOMMON_OPT="$(FCOMMON_OPT)" all
 endif
 endif
 endif
